webboot/webboot.go

```diff
@@ -23,9 +27,62 @@ var (
 	ipv6    = flag.Bool("ipv6", true, "use IPV6")
 )
 
+func wget(arg string) (io.ReadCloser, error) {
+	resp, err := http.Get(arg)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode != 200 {
+		return nil, fmt.Errorf("non-200 HTTP status: %d", resp.StatusCode)
+	}
+	return resp.Body, nil
+}
+
+func writer(read io.ReadCloser, fileName string) (string, error) {
+	w, err := os.Create(fileName)
+	if err != nil {
+		return "", err
+	}
+
+	defer w.Close()
+
+	_, err = io.Copy(w, read)
+	return fileName, err
+}
+
+func usage() {
+	log.Printf("Only one argument: an ISO link")
```

Comment:
I'm puzzled by this sequence of printfs in all these commits. Are you trying to keep the line short?
Also, "only one argument: an iso link" is printed BEFORE usage. You should generally try to keep the usage to one line.

Reply:
Yes, I was trying to keep the line short and simple, but however, I have gotten rid of the first printf() and then exit the user out of main if they do not provide me with the correct arguments.